product_extra_image_zipped_download/models/product_image.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import base64
+import mimetypes
+
+from odoo import api, fields, models
+from odoo.tools.mimetypes import get_extension, guess_mimetype
+
+
+class ProductImage(models.Model):
+    _inherit = "product.image"
+
+    img_name = fields.Char("Image Name")
+
+    def _get_image_name(self):
+        self.ensure_one()
+        if self.img_name:
+            return self.img_name
+        image_name = self.name
+        extension = get_extension(image_name)
+        if not extension:
+            image_data = base64.b64decode(self.image_1920)
+            mimetype = guess_mimetype(image_data)
+            extension = mimetypes.guess_extension(mimetype)
+            image_name = f"{image_name}{extension}"
+        # TODO: Assignment of self.id in the image name should be done in another module
+        return f"{self.id}{image_name}"
```

Comment:
```suggestion
        return image_name
```
@AungKoKoLin1997 
~~Please take the following actions:
・Create this PR at OCA.
・Create a custom module that changes the return value of _get_image_name to f"{self.id}{image_name}".~~

Reply:
@AungKoKoLin1997 
Yoshi-san and I discussed and decided to implement this module for the client as a custom module rather than submitting it to OCA.
If there are no other tasks and you have time, you can treat it as a low-priority task to make it OCA-compliant.
Therefore, there's no need for you to take any action on this PR at the moment.